src/renderers/dom/fiber/ReactDOMFiberEntry.js

```diff
@@ -646,7 +691,70 @@ function renderSubtreeIntoContainer(
   return DOMRenderer.getPublicRootInstance(root);
 }
 
+type PublicRoot = {
+  render(children: ReactNodeList, callback: ?() => mixed): void,
+  prerender(children: ReactNodeList): Work,
+  unmount(callback: ?() => mixed): void,
+
+  _reactRootContainer: *,
+  _getComponent: () => DOMContainer,
+};
+
+function PublicRootNode(
+  container: DOMContainer | (() => DOMContainer),
+  namespace: ?string,
+) {
+  if (typeof container === 'function') {
+    if (typeof namespace !== 'string') {
+      // Default to HTML namespace
+      namespace = DOMNamespaces.html;
+    }
+    this._reactRootContainer = DOMRenderer.createContainer(namespace);
+    this._getComponent = container;
+  } else {
+    // Assume this is a DOM container
+    const domContainer: DOMContainer = (container: any);
+    this._reactRootContainer = DOMRenderer.createContainer(domContainer);
+    this._getComponent = function() {
+      return domContainer;
+    };
+  }
+}
+PublicRootNode.prototype.render = function(
+  children: ReactNodeList,
+  callback: ?() => mixed,
+): void {
+  const work = DOMRenderer.updateRoot(children, this._reactRootContainer, null);
+  callback = callback === undefined ? null : callback;
+  work.then(() => {
+    work.commit();
+    if (callback !== null) {
+      (callback: any)();
+    }
+  });
+};
+PublicRootNode.prototype.prerender = function(children: ReactNodeList): Work {
+  return DOMRenderer.updateRoot(children, this._reactRootContainer, null);
+};
+PublicRootNode.prototype.unmount = function(callback) {
+  const work = DOMRenderer.updateRoot(null, this._reactRootContainer, null);
+  callback = callback === undefined ? null : callback;
+  work.then(() => {
+    work.commit();
+    if (callback !== null) {
+      (callback: any)();
+    }
+  });
+};
+
 var ReactDOMFiber = {
+  unstable_create(
+    container: DOMContainer | (() => DOMContainer),
+    namespace: ?string,
```

Comment:
Isn't this suppose to be `namespace ?: string`?

Reply:
Ooh, didn't know Flow treated those differently.